feat: SDP-9629: log more info on deployment error

mgoltzsche opened the pull request:
... by letting skaffold figure out and log the reason in case of a failed Helm chart deployment.

mgoltzsche commented on action.yml:
```diff
@@ -47,7 +47,9 @@ runs:
     - id: helm-install
       name: Install a Helm chart
-      uses: docker://public.ecr.aws/l7o7z1g8/actions/helm-install-action:1.0.0
+      # TODO: revert to using the public registry
+      #uses: docker://public.ecr.aws/l7o7z1g8/actions/helm-install-action:${{ action.scm.sha }}
+      uses: docker://registry.saas-dev.beescloud.com/staging/helm-install-action:${{ action.scm.sha }}
```
This needs to be solved before this change can be released. For now I am leaving it in here to show that the PR build works.

mgoltzsche replied:
After we saw the build succeed and the image from this PR was promoted to the public repo using the tag `1.1.0`, I changed the action.yml back to using the public registry and the new tag.

pinaki124 commented on action.yml:
```diff
@@ -75,14 +77,56 @@ runs:
+          echo "       Supported format: [0-9]+(s|m)" >&2
+          false
+        )
+        export TIMEOUT_SECONDS="$(echo "$TIMEOUT" | awk '{print match($1, "^([0-9]+)m$", m) ? substr($1, RSTART, RLENGTH-1)*60 : gensub("s$", "", "1", $1)}' -)"
```
This doesn't work if the user sets the `TIMEOUT` to `5m10s`, a format which `helm` supports:

```
      --timeout duration                           time to wait for any individual Kubernetes operation (like Jobs for hooks) (default 5m0s)
```

pinaki124 commented:
I also see all the install steps logging this message as a `Warning`:

```
time="2023-11-15T20:45:33Z" level=warning msg="Could not get docker context: starting command docker context inspect --format {{.Endpoints.docker.Host}}: exec: \"docker\": executable file not found in $PATH, falling back to the default docker host" subtask=-1 task=DevLoop
```
Is there anyway to not log this - I fear this may throw users off when they are trying to debug why a deployment failed.

pinaki124 commented:
I also see this message:

```
There is a new version (2.9.0) of Skaffold available.
```

Since this is our first version being released with `skaffold`, should we update to use the latest version?

mgoltzsche commented:
@pinaki124 I addressed your concerns now:
* To prevent skaffold from logging the docker-related warning, I added a fake `docker` script.
* I made the Action use the new skaffold 2.9.0 version.
* To support the full `timeout` input value format Helm supports, I added a `duration2seconds` bash script and corresponding bats tests.

Please re-review!

mgoltzsche commented:
I am closing this one in favour of #3 which merges into the new dev branch instead of main to avoid breaking things for users referring to the main branch.

mgoltzsche commented:
Reopening this again and closing #3 since I realized we cannot replace the helm-install@v1 usage with a pre-configured Action anymore within our organization and I assume nobody is using the Action from the main branch directly anyway but a pinned tag.
Thus, I am updating the image reference to point to the new public image version before merging this one...

mgoltzsche commented:
@jpbriend [pointed out](https://github.com/cloudbees-io/helm-install/pull/3#discussion_r1397473393) that there is bats container image that could be used here. I am also applying that to the PR...
